src/app/productConfig/productConfigModal.tpl.html

```diff
@@ -42,26 +50,33 @@ <h4 class="panel-title border-bottom-small" translate>
             <!--</div>-->
 
             <div data-toggle="buttons" ng-class="{'has-error': ($ctrl.itemConfigForm.amount | showErrors)}">
-              <label class="btn btn-radio" ng-click="$ctrl.changeAmount(a)" ng-class="{'active': $ctrl.itemConfig.amount === a}" ng-repeat-start="a in $ctrl.selectableAmounts">
+              <label class="btn btn-radio"
+                     ng-click="$ctrl.changeAmount(a)"
+                     ng-class="{'active': $ctrl.itemConfig.amount === a && !$ctrl.customInputActive}"
+                     ng-repeat-start="a in $ctrl.selectableAmounts">
                 <span class="number">${{a}}</span>
               </label>
               <span ng-repeat-end></span>
-              <label class="btn btn-default-form u-textLeft custom-amount">
+              <label
+                ng-class="{active: $ctrl.customInputActive}"
+                class="btn btn-default-form u-textLeft custom-amount">
                 <div class="form-group form-group-default u-inline">
                   <input class="form-control form-control-subtle number"
-                         type="number"
                          name="amount"
+                         type="text"
+                         step="1"
+                         ng-required="$ctrl.customInputActive"
                          ng-model="$ctrl.customAmount"
                          ng-change="$ctrl.changeCustomAmount($ctrl.customAmount)"
-                         ng-min="1"
-                         ng-max="10000000"
-                         ng-pattern="/^[0-9]+(\.[0-9]{1,2})?$/">
+                         ng-focus="$ctrl.customInputActive = true;"
+                         required>
                 </div>
               </label>
               <div role="alert" ng-messages="$ctrl.itemConfigForm.amount.$error" ng-if="($ctrl.itemConfigForm.amount | showErrors)">
-                <div class="help-block" ng-message="min" translate>Amount must be at least $1.00</div>
-                <div class="help-block" ng-message="max" translate>Amount must be less than $10,000,000.00</div>
-                <div class="help-block" ng-message="pattern" translate>Amount must be a valid number with no more than 2 decimal places</div>
+                <div class="help-block" ng-message="pattern" translate>Amount must be a valid number</div>
+                <div class="help-block" ng-message="required" translate>Amount must be not empty</div>
+                <div class="help-block" ng-message="minimum" translate>Amount must be more than $1.00</div>
```

Comment:
`Amount must be at least $1.00`? The greater than equal cases are always weird to phrase. Also should translation handle the localization of this static number or should we use https://docs.angularjs.org/api/ng/filter/currency?